[quant][decomposed] Add support for int32 for decomposed q/dq ops

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #89881

Summary:
att

Test Plan:
python test/test_quantization.py -k test_decomposed_quantize_per_tensor
python test/test_qunatization.py -k test_decomposed_dequantize_per_tensor

Reviewers:

Subscribers:

Tasks:

Tags:

cc @jianyuh @raghuramank100 @jamesr66a @vkuzo @jgong5 @Xia-Weiwen @leslie-fang-intel

jerryzh168 committed:
[quant][decomposed] Add support for int32 for decomposed q/dq ops

Summary:
att

Test Plan:
python test/test_quantization.py -k test_decomposed_quantize_per_tensor
python test/test_qunatization.py -k test_decomposed_dequantize_per_tensor

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

cccclai reviewed (approved):
LGTM!

cccclai commented on torch/ao/quantization/fx/_decomposed.py:
```diff
@@ -17,6 +17,9 @@ def _quant_min_max_bounds_check(quant_min, quant_max, dtype):
         quant_min_lower_bound = -128
         quant_max_upper_bound = 127
+    elif dtype == torch.int32:
+        quant_min_lower_bound = -2**31
```
```suggestion
        quant_min_lower_bound = -(2**31)
```

cccclai replied:
Also maybe define the constant somewhere?

jerryzh168 committed:
Update on "[quant][decomposed] Add support for int32 for decomposed q/dq ops"


Summary:
att

Test Plan:
python test/test_quantization.py -k test_decomposed_quantize_per_tensor
python test/test_qunatization.py -k test_decomposed_dequantize_per_tensor

Reviewers:

Subscribers:

Tasks:

Tags:

cc jianyuh raghuramank100 jamesr66a vkuzo jgong5 Xia-Weiwen leslie-fang-intel

[ghstack-poisoned]

jerryzh168 commented:
@pytorchbot rebase

jerryzh168 committed:
Update on "[quant][decomposed] Add support for int32 for decomposed q/dq ops"


Summary:
att

Test Plan:
python test/test_quantization.py -k test_decomposed_quantize_per_tensor
python test/test_qunatization.py -k test_decomposed_dequantize_per_tensor

Reviewers:

Subscribers:

Tasks:

Tags:

cc jianyuh raghuramank100 jamesr66a vkuzo jgong5 Xia-Weiwen leslie-fang-intel

[ghstack-poisoned]

jerryzh168 commented:
@pytorchbot merge
